### ISSUE TITLE
How To Use In A Class Component

### ISSUE THREAD

**Issue by ct-gilford** [asotog/fabricjs-react#8]
I wish to use it in a class component but I keep getting this error:

Line 161:33:  React Hook "useFabricJSEditor" cannot be called in a class component. React Hooks must be called in a React function component or a custom React Hook function  react-hooks/rules-of-hooks

So is it possible to use it in a class component?

**Comment by asotog** [comment]
no, you can't but you may workaround a bit, like wrap the useFabricJSEditor around a functional component and use that functional component from class component, acting as an intermediate layer, I have done that before for other projects

**Comment by ct-gilford** [comment]
> no, you can't but you may workaround a bit, like wrap the useFabricJSEditor around a functional component and use that functional component from class component, acting as an intermediate layer, I have done that before for other projects

But even with that, it won't help with what I wish to achieve, which is to trigger the editor to add a circle or something when clicking on a button from a different component. Even if I can get the reference of the class component that wrapped the functional component, I can't call the functional component to do something like FunctionalComponent.addCircle().

**Comment by ct-gilford** [comment]
> > no, you can't but you may workaround a bit, like wrap the useFabricJSEditor around a functional component and use that functional component from class component, acting as an intermediate layer, I have done that before for other projects
> 
> But even with that, it won't help with what I wish to achieve, which is to trigger the editor to add a circle or something when clicking on a button from a different component. Even if I can get the reference of the class component that wrapped the functional component, I can't call the functional component to do something like FunctionalComponent.addCircle().

@asotog I found a workaround, from the parent component, I create and pass down a new object as a prop. Then, in the child component, I populate the object with the internal functions like so `object.foo = foo;` where `foo` is the defined function inside the child component. Now, if I want to call any of the child component's internal functions, I can call `object.foo()`.

So I think you can close this issue. But I recommend adding some sample codes in the Readme on how to do this whether with an actual class component or with a workaround method like mine.

**Comment by asotog** [comment]
interesting, also I recommend trying to use `ref` instead from the class component, which is also possible to achieve the same you did, but this sounds more like extended implementation to what the actual component does @ct-gilford

**Comment by ct-gilford** [comment]
> interesting, also I recommend trying to use `ref` instead from the class component, which is also possible to achieve the same you did, but this sounds more like extended implementation to what the actual component does @ct-gilford.
> 
> From the class component, you do like `https://reactjs.org/docs/refs-and-the-dom.html#creating-refs` and from function component, you say like `ref.current = editor`

That would work too. Thanks for the suggestion!
